src/commands/init_dataset.rs

```diff
@@ -0,0 +1,26 @@
+use chrono::{Datelike, Utc};
+use crate::{config::E4EDMConfig, dataset::build_dataset};
+use anyhow::{Result, bail};
+use directories::ProjectDirs;
+
+use super::InitDatasetArgs;
+
+pub(crate) fn init_dataset(args: &InitDatasetArgs, config: &mut E4EDMConfig) -> Result<()> {
+    let dataset_name = format!("{year:04}.{month:02}.{day:02}.{project}.{location}", 
+        year = args.date.year(),
+        month = args.date.month(),
+        day = args.date.day(),
+        project = args.project,
+        location = args.location
+    );
+    let dataset_path = args.path.to_owned()
+        .unwrap_or(ProjectDirs::from("com", "Engineers For Exploration", "E4EDataManagement").unwrap().config_dir().to_path_buf())
+        .join(dataset_name.clone());
+
+    if config.datasets.contains_key(&dataset_name) {
+        bail!("Dataset with that name already exists!");
+    }
+    config.active_dataset = Some(dataset_name.clone());
+    config.datasets.insert(dataset_name, build_dataset(dataset_path, args.date.with_timezone(&Utc)));
```

Comment:
> Also, I don't know if it makes sense to store the UTC timestamp. My thought we can probably just get away with using the date (the `NaiveDate` type in `chrono`) since this the dataset is meant to have more coarse-grain information anyway.

@falafel72 I've made the changes in my most recent commit ([7b903e4](https://github.com/UCSD-E4E/e4e-data-management-rust/pull/13/commits/7b903e4510a8c1c2fac31e2b5661fab386975caf)) but wasn't sure if this changed something with `serde` since I removed one of the annotations while making this change, and I'm not familiar enough with it to know exactly what that line did. I'm aware @cgcha did some stuff with `serde` -- could you verify that what I've done is ok?